File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        st = []
        
        for c in s:
            if(c == '(' or c == '{' or c == '['):
                st.append(c)
            elif(len(st) > 0 and st[-1] == '(' and c == ')'): st.pop()
            elif(len(st) > 0 and st[-1] == '{' and c == '}'): st.pop()
            elif(len(st) > 0 and st[-1] == '[' and c == ']'): st.pop()
            else: return False
        
        if len(st) > 0: 
            return False
        return True

File: 20-valid-parentheses/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        return solve(s);
    }
private:
    bool solve(string s){
        stack<char> st;
        
        for(char c: s){
            if(c == '(' or c == '{' or c == '[')
                st.push(c);
            else if(!st.empty() and st.top() == '(' and c == ')') st.pop();
            else if(!st.empty() and st.top() == '{' and c == '}') st.pop();
            else if(!st.empty() and st.top() == '[' and c == ']') st.pop();
            else return false;
        }
        if(!st.empty()) return false;
        return true;
    }
};